### DISCUSSION TITLE
Issues encountered when retrieving state vectors that violated the constraints

### DISCUSSION THREAD

**Discussion by Jushan-Randy-Chen** [do-mpc/do-mpc#326] [Q&A] [answered]
Do you just want to pose a question or start a discussion? Please use the Discussions tab above.

Hi,

I followed the [FAQ ](https://www.do-mpc.com/en/latest/FAQ.html#feasibility-issues)page here to retrieve the potential state vectors that violated constraints in my ` mpc` setup (I've been getting the warning "...converged to local infeasibility... when running the simulation".  However, after I did exactly what's written on the documentation, a blank vector is returned 
```
lb_bound_violation = mpc.opt_x_num.cat
opt_labels = mpc.opt_x.labels()
labels_lb_viol =np.array(opt_labels)[np.where(lb_bound_violation)[0]]
labels_lb_viol
```
```
array([], dtype='<U22')
```

**Comment by 4flixt** [comment, marked as answer]
Hi @RandyChen233,

thank you for your question. This means than none of the bounds (lower and upper bounds for x and u) were violated. do-mpc includes additional constraints, however, that will be the cause of your problem. The constraints are more difficult to diagnose, however and include: The system dynamics (enforced through orthogonal collocation on finite elements) and any constraints you have introduced through ``mpc.set_nl_cons(...)``. 
To solve the problem you should first remove the latter or convert them to soft constraints. 
Also you might want to re-check your system model for any errors. There is no out-of-the-box solution for that. A good start is to perform simple simulations (e.g. with constant inputs) and see if it behaves as expected. 

Best,
4flixt
